sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkSQLCLIDriver.scala

```diff
@@ -81,11 +81,7 @@ private[hive] object SparkSQLCLIDriver extends Logging {
       System.exit(1)
     }
 
-    val cliConf = new HiveConf(classOf[SessionState])
-    // Override the location of the metastore since this is only used for local execution.
-    HiveUtils.newTemporaryConfiguration(useInMemoryDerby = false).foreach {
```

Comment:
This function not only constructs the warehouse path, it also create many other configs, why is it safe to just ignore em?

Reply:
before #SPARK-21428, cliSessionState will be left behind; now we will reuse it. this func is used to organize execution hive configs, but now cliSessionState has to talk with metastore.

Reply:
I don't think you have answered the question here - for example, the original code will update the `HiveConf.ConfVars.METASTORECONNECTURLKEY` property here, but you don't touch that after the change. I'm not confident to make that behavior change.

Reply:
Change this key to connect a dummy metastore instead of the real one？

Reply:
METASTORECONNECTURLKEY connects derby by default, we may set this key in hive-site.xml to talk with metastore. hiveclient will use the state init here, I don't think we should beinit hive conf as the old way.

Reply:
@jiangxb1987 sorry for not @ you， please take a look again